### PR TITLE
(feat) - cleanup changing refs correctly

### PR DESCRIPTION
**Pull request by JoviDeCroock**
Fixes: https://github.com/preactjs/preact/issues/2052

This differentiates in one aspect and that is that we do
- call with null
- call with value

instead of
nullcalls
valuecalls

### CODE REVIEW

File: src/diff/children.js [hunk 1]
```diff
@@ -115,7 +115,8 @@ export function diffChildren(
 					(refs || (refs = [])).push(
```
Comment (andrewiggins) [inline src/diff/children.js]: What's the size if call `applyRef` twice?
```js
if ((j = childVNode.ref) && oldVNode.ref != j) {
   if (!refs) refs = [];
   if (oldVNode.ref) refs.push(oldVNode.ref, null, childVNode);
   refs.push(j, childVNode._component || ... );
}
```
Or something similar...

The if statement in `diffChildren` already checks if the refs are different so ideally we could reuse that `if` and not duplicate it in `applyRef` (if I'm reading this code correctly).

File: src/diff/index.js [hunk 2]
```diff
@@ -381,9 +381,9 @@ function diffElementNodes(
  * @param {any} value
  * @param {import('../internal').VNode} vnode
  */
-export function applyRef(ref, value, vnode) {
+export function applyRef(ref, value, vnode, oldRef) {
```
Comment (sventschui) [inline src/diff/index.js]: On my phone so maybe I’m a bit lost here but what’s this argument for?

Comment (JoviDeCroock) [reply src/diff/index.js]: oops, me being dumb good spot 👍

**Comment by JoviDeCroock** [comment]
> Thanks! LGTM 🎉

No no, Thank YOU for the idea on how to reduce size, hero!

**Pull request merged**